src/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/BlobUtilities.cs

```diff
@@ -271,5 +287,60 @@ internal static void ValidateRange(int bufferLength, int start, int byteCount)
                 Throw.ArgumentOutOfRange(nameof(byteCount));
             }
         }
+
+        internal static int GetUserStringByteLength(int characterCount)
+        {
+            return characterCount * 2 + 1;
+        }
+
+        internal static byte GetUserStringTrailingByte(string str)
+        {
+            // Write out a trailing byte indicating if the string is really quite simple
```

Comment:
It would be better to quote the spec here than just saying "quite simple":

II.24.2.4 "This final byte holds the value 1 if and only if any UTF16 character within the string has any bit set in its top byte, or its low byte is any of the following: 0x01– 0x08, 0x0E–0x1F, 0x27, 0x2D, 0x7F.  Otherwise, it holds 0. The 1 signifies Unicode characters that require handling beyond that normally provided for 8-bit encoding sets."